src/pages/lancer-un-defi/ecologique/4.tsx

```diff
@@ -38,71 +52,129 @@ const DefiEcoStep4 = () => {
+          activeStep={3}
+          errorSteps={errorSteps}
         />
         <div className="width-900">
-          <h1>Quel défi voulez-vous lancer aux Pelicopains ?</h1>
-          <div style={{ marginTop: '1rem' }}>
-            {ECO_DEFIS.map((t, index) => (
-              <ThemeChoiceButton key={index} label={t.title} description={t.description} onClick={onClick(index)} />
-            ))}
-            <ThemeChoiceButton
-              isOpen={otherOpen}
+          <h1>Pré-visualisez votre défi{!isEdit && ', et publiez-la'}</h1>
+          <p className="text" style={{ fontSize: '1.1rem' }}>
+            Voici la pré-visualisation de votre défi.
+            {isEdit
+              ? " Vous pouvez le modifier à l'étape précédente, et enregistrer vos changements ici."
+              : ' Vous pouvez le modifier, et quand vous êtes prêts : publiez-le dans votre village-monde !'}
+          </p>
+          {isEdit ? (
+            <div style={{ width: '100%', display: 'flex', justifyContent: 'space-between', margin: '1rem 0' }}>
+              <Link href="/lancer-un-defi/ecologique/4" passHref>
+                <Button component="a" color="secondary" variant="contained" href="/lancer-un-defi/ecologique/4">
+                  {"Modifier à l'étape précédente"}
+                </Button>
+              </Link>
+              <Button variant="outlined" color="primary" onClick={onPublish}>
+                Enregistrer les changements
+              </Button>
+            </div>
+          ) : (
+            <div style={{ width: '100%', textAlign: 'right', margin: '1rem 0' }}>
+              {!isValid && (
+                <p>
+                  <b>Avant de publier votre présentation, il faut corriger les étapes incomplètes, marquées en orange.</b>
+                </p>
+              )}
+              <div style={{ width: '100%', textAlign: 'right', margin: '1rem 0' }}>
+                <Button variant="outlined" color="primary" onClick={onPublish} disabled={!isValid}>
+                  Publier
+                </Button>
+              </div>
+            </div>
+          )}
+
+          {!isEdit && activity.responseActivityId === null && <ActivityLink url={`/lancer-un-defi/ecologique/1?edit=${activity.id}`} />}
+          {responseActivity !== null && (
+            <>
+              <span className="text text--small text--success">Défi en réaction à {REACTIONS[responseActivity.type]}</span>
+              <div className="preview-block">
+                {!isEdit && (
+                  <EditButton
+                    onClick={() => {
+                      router.push(`/lancer-un-defi/ecologique/1?edit=${activity.id}`);
+                    }}
+                    status={'success'}
+                    style={{ position: 'absolute', top: '0.5rem', right: '0.5rem' }}
+                  />
+                )}
+                <Activities activities={[responseActivity]} noButtons />
+              </div>
+            </>
+          )}
```

Comment:
Same as what I said on another of your PR. The response activity stuff is not needed anymore for the activities. (Expect for the Reaction activity which is dedicated to it.) So this can be removed.

Reply:
I will do it in global PR.